docs/prow/templates.md

```diff
@@ -0,0 +1,74 @@
+# Templates
+
+Jobs and Prow configuration are generated from templates that are in `templates` directory. There are following items there:
+
+- `config.yaml` - config file that describes files to generate
+- `templates` - directory to gather all template files
+
+### Config structure
+
+Config file is expected to have two keys:
+
+- `global` - a map of values available for all templates
+- `templates` - a list of files to generate
+
+Complete structure can be found [here](../../development/tools/cmd/rendertemplates/main.go).
+
+Other then that `config.yaml` uses YAML techniques for better organization, so normal YAML rules apply.
```

Comment:
```suggestion
Other then that, `config.yaml` uses YAML techniques for better organization, so normal YAML rules apply.
```
This sounds somewhat too colloquial... 
Maybe change it to something like:
> `config.yaml` uses YAML techniques for better organization. If you use YAML, you are already familiar with the rules. 

?

Reply:
On the second thought, if you see it is a YAML you may expect all the YAML things to be there, so I removed the sentence completely.